### PR TITLE
[All hosts](deployment) Add article on admin consent

### PR DESCRIPTION
**Pull request by davidchesnut**
This article is a how to on adding admin consent to your Office add-in deployment when updating. You can use either an admin consent URI, or the unified manifest.
also some updates to the FAQ and other articles based on recent activity.

### CODE REVIEW

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+To publish updates there are three things to consider.
+
+1. How to deploy changes made in the add-in code to your web server.
+1. How to obtain admin consent to the Microsoft Graph scopes.
+1. How to deploy updates in your manifest.
+
+## Deploy updates to your add-in code
+
+Once you've updated and tested your add-in code, you'll need to deploy those updates to your web server. Follow your own update process (such as staging and production). Once deployed, all users of your add-in will see the changes and start using the updated add-in code. There is no need for admins or users to take any actions to see the updates.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:22]:
```suggestion
Once you've updated and tested your add-in code, you'll need to deploy them to your web server. Follow your own update process (such as staging and production). Once deployed, all users of your add-in will see the changes and start using the updated add-in code. There is no need for admins or users to take any actions to see the updates.
```

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+Any Microsoft Graph scopes used by your add-in require consent from either the user or the admin of a tenant. If the admin doesn't consent, then when your add-in requests an access token through MSAL, the user will be prompted to provide consent. In general the best user experience is to avoid prompting users for consent at all. You can have the admin consent for the entire tenant.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:24]:
```suggestion
Any Microsoft Graph scopes used by your add-in require consent from either the user or the admin of a tenant. If the admin doesn't consent, the user will be prompted to provide consent when your add-in requests an access token through MSAL. For the best user experience, avoid prompting users for consent at all. Instead, ask your admin to provide consent for the entire tenant.
```

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+```html
+https://login.microsoftonline.com/organizations/v2.0/adminconsent?client_id=c6c1f32b-5e55-4997-881a-753cc1d563b7&scope=https://graph.microsoft.com/User.Read https://graph.microsoft.com/Files.Read&redirect_uri=https://localhost:3000/consentRedirect.html
+```
+
+> [!IMPORTANT]
+> The redirect page must be added to the list of SPA redirects in your app registration along with the `brk-multihub` redirect or the admin consent URI will fail.
+
+## Get admin consent via the unified manifest
+
+You can also get admin consent as an automatic part of the deployment workflow when your add-in is deployed. To do this you add the `webApplicationInfo` property to your unified manifest. Then the admin deploys the updated manifest, either through central deployment, or from an update through Microsoft AppSource. When the admin deploys the updated manifest they are automatically prompted to consent to the scopes required by the add-in. If they don't consent, the updated add-in will not deploy.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:55]:
```suggestion
You can also get admin consent as an automatic part of the deployment workflow when your add-in is deployed. To do this, add the `webApplicationInfo` property to your unified manifest. Then the admin deploys the updated manifest, either through central deployment, or from an update through Microsoft AppSource. When the admin deploys the updated manifest, they are automatically prompted to consent to the scopes required by the add-in. If they don't consent, the updated add-in will not deploy.
```

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    :::image type="content" source="../images/azure-portal-request-api-permissions-graph.png" alt-text="The Request API permissions pane with Microsoft Graph button.":::
+
+1. Select **Delegated permissions**.
+
+    :::image type="content" source="../images/azure-portal-request-api-permissions-delegated.png" alt-text="The Request API permissions pane with delegated permissions button.":::
+
+1. In the **Select permissions** search box, search for the permissions your add-in needs. For example, for an Outlook add-in, you might use `profile`, `openid`, `Files.ReadWrite`, and `Mail.Read`.
+
+1. Select the checkbox for each permission as it appears. Note that the permissions will not remain visible in the list as you select each one. After selecting the permissions that your add-in needs, select **Add permissions**.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:85]: What happens if they wish to deselect a permission they've chosen if it's no longer visible? Just cancel and start over?

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        "id": "a92ace55-9daf-47bc-84e9-065e9a6e70e3",
+        "resource": "https://contoso.com"
+    },
+```
+
+### Deploy the updated manifest
+
+If you deployed your add-in through Microsoft AppSource, you'll need to submit your updated unified manifest for approval. For more information, see [Microsoft AppSource submission FAQ](/partner-center/marketplace-offers/appsource-submission-faq).
+
+If you deployed you add-in by providing the manifest to admins for central deployment, you'll need to provide admins with an updated [app package](/microsoftteams/platform/concepts/build-and-test/apps-package) that contains the updated manifest.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:104]:
```suggestion
If you deployed your add-in by providing the manifest to admins for central deployment, you'll need to provide admins with an updated [app package](/microsoftteams/platform/concepts/build-and-test/apps-package) that contains the updated manifest.
```

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+Once you've updated and tested your add-in code, you'll need to deploy those updates to your web server. Follow your own update process (such as staging and production). Once deployed, all users of your add-in will see the changes and start using the updated add-in code. There is no need for admins or users to take any actions to see the updates.
+
+Any Microsoft Graph scopes used by your add-in require consent from either the user or the admin of a tenant. If the admin doesn't consent, then when your add-in requests an access token through MSAL, the user will be prompted to provide consent. In general the best user experience is to avoid prompting users for consent at all. You can have the admin consent for the entire tenant.
+
+There are two ways to get admin consent; use an admin consent URI, or use the unified manifest.
+
+## Get admin consent via admin consent URI
+
+You can get admin consent by providing an admin consent URI. It provides a link that an admin can select. A dialog box prompts them to sign in with admin credentials, and consent to the Microsoft Graph scopes your add-in requires.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:30]: Is there an example of the dialog box prompt you can provide here?

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+- `redirect_uri`: A redirect page for when consent is completed. The Microsoft identity platform will redirect to this page after an admin consents. The redirect page is sent a success or fail JSON message as specified in [Admin consent on the Microsoft identity platform](/entra/identity-platform/v2-admin-consent). On the page you can indicate the consent status to the admin as well as provide more information or next steps about your add-in.
+
+The following admin URI example shows how to specify the `User.Read` and `Files.Read` scopes and redirect to a page on your web server named `consentRedirect.html`.
+
+```html
+https://login.microsoftonline.com/organizations/v2.0/adminconsent?client_id=c6c1f32b-5e55-4997-881a-753cc1d563b7&scope=https://graph.microsoft.com/User.Read https://graph.microsoft.com/Files.Read&redirect_uri=https://localhost:3000/consentRedirect.html
+```
+
+> [!IMPORTANT]
+> The redirect page must be added to the list of SPA redirects in your app registration along with the `brk-multihub` redirect or the admin consent URI will fail.
```
Comment (lindalu-MSFT) [inline docs/publish/publish-nested-app-auth-add-in.md:51]:
```suggestion
> The redirect page must be added to the list of single-page application (SPA) redirects in your app registration along with the `brk-multihub` redirect or the admin consent URI will fail.
```

File: docs/publish/publish-nested-app-auth-add-in.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+> The redirect page must be added to the list of single-page application (SPA) redirects in your app registration along with the `brk-multihub` redirect or the admin consent URI will fail.
+
+The following admin URI example shows how to specify the `openid`, `profile`, `offline_access`, `user.read` and `files.read` scopes and redirect to a page on your web server named `consentRedirect.html`.
+
+```html
+https://login.microsoftonline.com/organizations/v2.0/adminconsent?client_id=63e62b68-c5c7-48f9-82bf-8c41d5637b49&scope=offline_access openid profile user.read files.read&redirect_uri=https://localhost:3000/consentRedirect.html
+```
+
+> [!NOTE]
+> In requests to the authorization, token or consent endpoints for the Microsoft identity platform, if the resource identifier is omitted in the scope parameter, the resource is assumed to be Microsoft Graph. For example, `User.Read` is equivalent to `https://graph.microsoft.com/User.Read`.
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/publish/publish-nested-app-auth-add-in.md]: fixed

**Pull request merged**